### DISCUSSION TITLE
How to update the datasource automatically

### DISCUSSION THREAD

**Discussion by karah9** [Power-Components/livewire-powergrid#305] [Q&A]
I want to update my datasource depending on the select from livewire component

**Comment by luanfreitasdev** [comment]
I didn't understand your question

**Comment by insulae** [comment]
I'm having the same problem and I don't know how to solve it, I refresh the table using 
`pg:eventRefresh-default `
but I don't know how to update the data that comes from outside powergrid.
With other components I usually do it like this:
`<livewire:table :area_id="$area_id" >`
but in powergrid it initializes $area_id but never updates it, could you solve it?

**Reply by elsoftpy** [reply]
Hi!

Perhaps you've already solved the issue but I just wanted to share my solution to anyone looking to update the datasource while refreshing the table component.

In my livewire component I dispatch an event in order to pass the update datasource like this: 
```
public function addDetail()
{
    $this->dispatch('refreshDatasource', detail: $this->detail)
       ->to(DetailTableComponent::class);
}
``` 

In my table component I set a new listener to listen for the ```refreshDatasource``` event, then inside the function I update the datasource  and  I dispatch the ```pg:eventRefresh``` event.

```
public array $detail; 
public string $tableName = 'DetailTable'; 

public function datasource(): ?Collection
{
    return collect($this->detail);
}
...
#[On('refreshDatasource')]
    public function refreshDatasource($detail): void
    {
        $this->detail = $detail;

        $this->dispatch('pg:eventRefresh-' . $this->tableName);
    }
```

**Reply by ohmeng1984** [reply]
this is great but how to refresh your datasource using eloquent builder?
